Criação do parametro opcional DigestMethod #673

gui-abrantes opened the pull request:
Criação do parametro opcional DigestMethod com valor padrao de "http://www.w3.org/2000/09/xmldsig#sha1" para a função de assinar a nfe.

Mesmo sendo o padrao hj da NFe SHA1, como todos entendemos nosso governo, vai q ele resolve mudar...já fica previsto essa mudança de padrão.

gui-abrantes committed:
Criação do parametro opcional DigestMethod com valor padrao de "http://www.w3.org/2000/09/xmldsig#sha1" para a função de assinar a nfe.

Mesmo sendo o padrao hj da NFe SHA1, como todos entendemos nosso governo, vai q ele resolve mudar...já fica previsto essa mudança de padrão.

gui-abrantes commented:
@rodrigordi Realmente, esqueci desse detalhe, mas coloquei a linha que vc sugeriu e deu o erro:
"O SignatureDescription não pôde ser criado para o algoritmo de assinatura fornecido."

E portanto gostaria de invalidar essa pull request, pq realmente está com erro! Por enquanto pela solução de editar o app.config mesmo pq o meu prazo esta meio curto! rsrs....e futuramente podemos melhorar

rodrigordi commented:
@gui-abrantes , tranquilo . voce deve ter colocado o mesmo valor para os dois.
"http://www.w3.org/2000/09/xmldsig#rsa-sha1"; <----- signmethod
"http://www.w3.org/2000/09/xmldsig#sha1"; <------- digest value

gui-abrantes committed:
Mudança na classe ConfiguracaoServico com as as novas propriedades:
-SignatureMethodSignedXml;
-DigestMethodReference;

E passagem destes valores novos para o assinador, assinar de acordo com a configuração;

gui-abrantes commented:
@rodrigordi valeu pela dica....agora está funcionando! Pensei em um forma melhor e coloquei a configuração na classe de configuração, com as seguintes propriedades:
-SignatureMethodSignedXml com valor padrao http://www.w3.org/2000/09/xmldsig#rsa-sha1;
-DigestMethodReference com valor padrao http://www.w3.org/2000/09/xmldsig#sha1;

Entao ou seja, ao instanciar já tem esses valores predefinidos, caso deseje mudar, basta mudar as propriedades desta classe.

Att,
Guilherme

gui-abrantes committed:
Adequacao da propriedades de DigestMethodReference e SignatureMethodSignedXml para dentro da classe ConfiguracaoCertificado

gui-abrantes committed:
Merge remote-tracking branch 'mudancas/master'

gui-abrantes committed:
Inclusao propriedade refCTe

gui-abrantes committed:
Merge remote-tracking branch 'mudancas/master'

robertorp commented on NFe.AppTeste/MainWindow.xaml.cs:
```diff
@@ -429,9 +429,9 @@ private void BtnCriareEnviar3_Click(object sender, RoutedEventArgs e)
+                //var retornoEnvio = servicoNFe.NFeAutorizacao(Convert.ToInt32(lote), IndicadorSincronizacao.Assincrono, new List<Classes.NFe> {_nfe}, false/*Envia a mensagem compactada para a SEFAZ*/);
                 //Para consumir o serviço de forma síncrona, use a linha abaixo:
-                //var retornoEnvio = servicoNFe.NFeAutorizacao(Convert.ToInt32(lote), IndicadorSincronizacao.Sincrono, new List<Classes.NFe> { _nfe }, true/*Envia a mensagem compactada para a SEFAZ*/);
+                var retornoEnvio = servicoNFe.NFeAutorizacao(Convert.ToInt32(lote), IndicadorSincronizacao.Sincrono, new List<Classes.NFe> { _nfe }, true/*Envia a mensagem compactada para a SEFAZ*/);
```
Poderia voltar ao estado normal? este com "//"

robertorp commented on NFe.AppTeste/MainWindow.xaml.cs:
```diff
@@ -429,9 +429,9 @@ private void BtnCriareEnviar3_Click(object sender, RoutedEventArgs e)
 
                 var servicoNFe = new ServicosNFe(_configuracoes.CfgServico);
-                var retornoEnvio = servicoNFe.NFeAutorizacao(Convert.ToInt32(lote), IndicadorSincronizacao.Assincrono, new List<Classes.NFe> {_nfe}, false/*Envia a mensagem compactada para a SEFAZ*/);
+                //var retornoEnvio = servicoNFe.NFeAutorizacao(Convert.ToInt32(lote), IndicadorSincronizacao.Assincrono, new List<Classes.NFe> {_nfe}, false/*Envia a mensagem compactada para a SEFAZ*/);
```
Poderia voltar ao estado normal? este sem o "//" comentário

robertorp commented on NFe.Utils/Evento/Extevento.cs:
```diff
@@ -56,13 +56,13 @@ public static string ObterXmlString(this evento pedEvento)
         /// <param name="certificadoDigital">Informe o certificado digital, se já possuir esse em cache, evitando novo acesso ao certificado</param>
         /// <returns>Retorna um objeto do tipo evento assinado</returns>
-        public static evento Assina(this evento evento, X509Certificate2 certificadoDigital)
+        public static evento Assina(this evento evento, X509Certificate2 certificadoDigital, string SignatureMethodSignedXml = "http://www.w3.org/2000/09/xmldsig#rsa-sha1", string DigestMethodReference = "http://www.w3.org/2000/09/xmldsig#sha1")
```
poderia colocar os parâmetros com iniciais minusculas?

robertorp commented on NFe.Utils/Assinatura/Assinador.cs:
```diff
@@ -67,7 +67,7 @@ public static class Assinador
         /// <param name="manterDadosEmCache">Validador para manter o certificado em cache</param>
         /// <returns>Retorna um objeto do tipo Classes.Assinatura.Signature, contendo a assinatura do objeto passado como parâmetro</returns>
-        public static Signature ObterAssinatura<T>(T objeto, string id, X509Certificate2 certificadoDigital, bool manterDadosEmCache = false) where T : class
+        public static Signature ObterAssinatura<T>(T objeto, string id, X509Certificate2 certificadoDigital, bool manterDadosEmCache = false, string signaturemethod = "http://www.w3.org/2000/09/xmldsig#rsa-sha1", string digestmethod = "http://www.w3.org/2000/09/xmldsig#sha1") where T : class
```
nome dos parâmetros poderia colocar
signatureMethod
digestMethod

robertorp commented on NFe.Utils/Inutilizacao/ExtinutNFe.cs:
```diff
@@ -67,13 +67,13 @@ public static string ObterXmlString(this inutNFe pedInutilizacao)
         /// <param name="certificadoDigital">Informe o certificado digital, se já possuir esse em cache, evitando novo acesso ao certificado</param>
         /// <returns>Retorna um objeto do tipo inutNFe assinado</returns>
-        public static inutNFe Assina(this inutNFe inutNFe, X509Certificate2 certificadoDigital)
+        public static inutNFe Assina(this inutNFe inutNFe, X509Certificate2 certificadoDigital, string SignatureMethodSignedXml = "http://www.w3.org/2000/09/xmldsig#rsa-sha1", string DigestMethodReference = "http://www.w3.org/2000/09/xmldsig#sha1")
```
Poderia colocar os nomes dos parâmetros para iniciar com minuscula?

gui-abrantes committed:
Ajustes e Padronização

robertorp commented:
@gui-abrantes  obrigado!
